languages/thingtalk/en/parameters.genie

```diff
@@ -116,6 +116,36 @@ where_question = {
     ) => C.makeSingleFieldProjection('table', Type.Location, table, 'geo');
 }
 
+boolean_projection = {
+    // npp
+    'does' ('the' | '' | 'any') table:with_filtered_table 'have' p:npp_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table 'with' p:npp_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // avp
+    'does' ('the' | '' | 'any') table:with_filtered_table p:avp_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that'|'') p:avp_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // pvp
+    'is' ('the' | '') table:with_filtered_table p:pvp_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that is'|'') p:pvp_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // preposition
+    'is' ('the' | '') table:with_filtered_table p:preposition_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that is'|'') p:preposition_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // npi
+    'is' ('the' | '') table:with_filtered_table p:npi_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that is'|'') p:npi_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // npv
+    'is' ('the' | '') table:with_filtered_table p:npv_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that is'|'') p:npv_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+
+    // apv
+    'is' ('the' | '') table:with_filtered_table p:apv_boolean_projection => C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+    'is there' ('any' | '') table:with_filtered_table ('that is'|'') p:apv_boolean_projection =>  C.makeSingleFieldProjection('table', Type.Boolean, table, p.name);
+}
```

Comment:
How does this compare with "boolean_user_search_question" (which is a coref question)?

Reply:
These two are still for unknown entities. the second one is clearer as it is just doing a search. the first one is to generate things like "is the restaurants in palo alto Chinese".

Reply:
No yeah the search stuff I understand, that's not projection and the grammar is different (it would match "negative_proposal_reply" from the refinement templates). I just wonder if we can unify the truly boolean templates, at least grammar-wise. Anyway not important right now.